File: backend/app/models/base/timeslot.py
from pydantic import BaseModel, Field
from datetime import date, time


class TimeSlot(BaseModel):
    slot_date: date
    start_time: time
    end_time: time


class TimeSlotDay(BaseModel):
    day: int = Field(ge=0, le=6)
    start_time: time
    end_time: time

File: frontend/src/lib/types/base/timeslot.ts
import { z } from "zod";

export const TimeSlotSchema = z.object({
    stot_date: z.string().date(),
    start_time: z.string().time(),
    end_time: z.string().time()
})

export const TimeSlotDaySchema = z.object({
    day: z.number().min(0).max(6),
    start_time: z.string().time(),
    end_time: z.string().time()
})

export type TimeSlot = z.infer<typeof TimeSlotSchema>;
export type TimeSlotDay = z.infer<typeof TimeSlotDaySchema>;
